flux-typeck/src/type_env/paths_tree.rs

```diff
@@ -160,7 +162,10 @@ impl PathsTree {
                             TyKind::Ref(mode, ty) => {
                                 return self.lookup_place_iter_ty(rcx, gen, *mode, ty, place_proj);
                             }
-                            _ => panic!("YIKES: {elem:?} {ty:?}"),
+                            TyKind::Indexed(bty, _) | TyKind::Exists(bty, _) if bty.is_box() => {
+                                return self.lookup_place_box_ty(bty);
+                            }
```

Comment:
We could match on the `bty` to avoid the panic in `lookup_place_box_ty`, but I don't have strong feelings for it. I always feel this type of deep matching sounds good conceptually, but not it doesn't look so good once you write it...

```suggestion
                            TyKind::Indexed(BaseTy::Adt(def, substs), _) | TyKind::Exists(BaseTy::Adt(def, substs), _) if def.is_box() => {
                                return LookupResult::Box(substs[0].clone());
                            }
```